### PR TITLE
[Swift] Support evaluating of `$0`/`$1` etc.. inside a closure.

### PR DESCRIPTION
**Pull request by dcci**
<rdar://problem/20719448>

### CODE REVIEW

File: source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp [hunk 1]
```diff
@@ -1184,13 +1184,8 @@ static swift::ASTContext *SetupASTContext(
...
     swift_ast_context->GetLanguageOptions().EnableObjCInterop = false;
 
-  if (repl || playground) {
-    swift_ast_context->GetLanguageOptions().Playground = true;
-    swift_ast_context->GetIRGenOptions().Playground = true;
-  } else {
-    swift_ast_context->GetLanguageOptions().Playground = true;
-    swift_ast_context->GetIRGenOptions().Playground = false;
-  }
+  swift_ast_context->GetLanguageOptions().Playground = repl || playground;
```
Comment (dcci) [inline source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp:1187]: This was a typo that prevented the feature to work correctly.

**Comment by dcci** [comment]
Compiler bits here: 
https://github.com/apple/swift/pull/19808

File: packages/Python/lldbsuite/test/lang/swift/closure_shortcuts/TestClosureShortcuts.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# TestAccelerateSIMD.py
```
Comment (jimingham) [inline packages/Python/lldbsuite/test/lang/swift/closure_shortcuts/TestClosureShortcuts.py]: That's not the right name...

**Review by jimingham — COMMENTED** [review]
With this patch in place, can I make $0 variables in random contexts.  What happens if I do:

(lldb) expression let $0 = 5

not in a closure context?  

And what kind of error do you get if you do:

(lldb) expression $0

in a non-closure context.  Currently this says:

error: anonymous closure argument not contained in a closure

it would be nice if we didn't lose that error and just let people make these variables.  It would be even worse if having made one  accidentally it collided with a real $0 variable...

If the behaviors in those cases are reasonable, then this seems great.  

If the answer is you can make $0 variables and they cause trouble but why would you ever do that...  Then I guess that's okay, it's more important to get this working.  But I'm a little sad.

File: packages/Python/lldbsuite/test/lang/swift/closure_shortcuts/TestClosureShortcuts.py (hunk 2 above)
Comment (dcci) [reply packages/Python/lldbsuite/test/lang/swift/closure_shortcuts/TestClosureShortcuts.py]: Thanks!

**Comment by dcci** [comment]
Thanks for your comments, Jim.

To reply to your questions:

```
(lldb) expression $0
error: use of undeclared identifier '$0'
```

(which seems reasonable to me). what do you think? I'm afraid there's not very much we can do here because `lldb` and `swift` have a fundamental namespace clash here. Variable names starting with `$` I think are really unlikely that maybe it's worth giving this one a shot. I'm not particularly happy with this solution either, but considering the number of reports we get about this being broken, it's probably OK to penalize other cases to get this one working.

**Comment by dcci** [comment]
Please test with following PR:
https://github.com/apple/swift/pull/19808

@swift-ci Please test

**Comment by jimingham** [comment]
We also checked by hand that if you do:

(lldb) expr let $0 = 100

which will succeed with this patch, and then stop in a frame that has a real $0 in it, we find the closure automatic variable $0 and not the one we defined in the expression. It might be a good idea to add a test for that so we make sure the lookup rules don't change and make that fail.

Other than that, this looks good to me.

**Comment by dcci** [comment]
Please test with following PR:
https://github.com/apple/swift/pull/19965

@swift-ci Please test

**Pull request merged**